src/lib.rs

```diff
@@ -45,49 +47,40 @@ impl<T, A, E: graph::Event> QueueHandler<T, A, E> {
         self.on(ev, handler);
         self
     }
+
+    fn handle_events(
+        handlers: &mut HandlerMap<T, A, E>,
+        events: &[E],
+        obj: &mut T,
+        additional: &mut A,
+    ) {
+        for event in events {
+            if let Some(handler) = handlers.get_mut(event.get_key()) {
+                (*handler)(obj, additional, event.clone());
+            }
+        }
+    }
 }
 
 impl<T, A, E: graph::Event> graph::DynQueueHandler<T, A> for QueueHandler<T, A, E> {
     fn update(&mut self, obj: &mut T, additional: &mut A) {
         let handlers = &mut self.handlers;
-        self.listener.with(|events| {
-            for event in events {
-                if let Some(handler) = handlers.get_mut(event.get_key()) {
-                    (*handler)(obj, additional, event.clone());
-                }
-            }
-        });
+        self.listener
+            .with(move |events| Self::handle_events(handlers, events, obj, additional));
     }
 
     fn update_n(&mut self, n: usize, obj: &mut T, additional: &mut A) {
         let handlers = &mut self.handlers;
-        self.listener.with_n(n, |events| {
-            for event in events {
-                if let Some(handler) = handlers.get_mut(event.get_key()) {
-                    (*handler)(obj, additional, event.clone());
-                }
-            }
+        self.listener.with_n(n, move |events| {
+            Self::handle_events(handlers, events, obj, additional)
         });
     }
 }
 
 /// A handler convertible to `Any`.
-pub trait AnyHandler<T, A>: graph::DynQueueHandler<T, A> {
-    fn as_any(&self) -> &dyn std::any::Any;
-    fn as_any_mut(&mut self) -> &mut dyn std::any::Any;
-}
-
-impl<T: 'static, A: 'static, E: graph::Event + 'static> AnyHandler<T, A> for QueueHandler<T, A, E> {
-    #[inline]
-    fn as_any(&self) -> &dyn std::any::Any {
-        self
-    }
+pub trait AnyHandler<T, A>: graph::DynQueueHandler<T, A> + 'static {}
```

Comment:
I'm not sure how this can act in place of `as_any/as_any_mut`. As far as I know, the `'static` isn't sufficient to cast `&dyn AnyHandler` to `&dyn Any` and hence downcast to a concrete `QueueHandler`.